src/clientlib/cl_subscription_manager.c

```diff
@@ -1447,13 +1447,24 @@ cl_sm_get_server_socket_filename(cl_sm_ctx_t *sm_ctx, const char *module_name, c
     snprintf(pid_str, 20, "%d", getpid());
     strncat(path, pid_str, PATH_MAX - strlen(path) - 1);
 
+#ifdef HAVE_MKSTEMPS
     /* append temporary file name part */
     strncat(path, ".XXXXXX.sock", PATH_MAX - strlen(path) - 1);
     fd = mkstemps(path, 5);
     if (-1 != fd) {
         close(fd);
         unlink(path);
     }
+#else
+    /* append temporary file name part */
+    strncat(path, ".XXXXXX", PATH_MAX - strlen(path) - 1);
+    fd = mkstemp(path);
+    if (-1 != fd) {
```

Comment:
I know it was like this before but it is not correct, so when you are modifying this code, it would be best if we added a correct error check. If `fd == -1` the whole function should fail I think.